packages/commerce-sdk-react/src/auth/storage.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import Cookies from 'js-cookie'
+
+export abstract class BaseStorage {
+    abstract set(key: string, value: string, options?: any): void
+    abstract get(key: string): string
+    abstract delete(key: string): void
+}
```

Comment:
```suggestion
export interface BaseStorage {
    set(key: string, value: string, options?: any): void
    get(key: string): string
    delete(key: string): void
}
```
An abstract class that doesn't implement anything is basically just an interface with extra bloat: it still exists at runtime, but doesn't do anything because it has no methods.

To validate that a class still implements an interface, just do `class CookieStorage implements BaseStorage`.

---

(To be pedantic, having it exist at runtime means you could do `storage instanceof BaseStorage`, but we don't do that and `storage instanceof CookieStorage || storage instanceof LocalStorage` is just as easy, if we ever need to.)

Reply:
✅